utils/debug/Debug.js

```diff
@@ -0,0 +1,128 @@
+
+    // line graph for nb elements
+    const nbObjectsCanvas = document.createElement('canvas');
+    nbObjectsCanvas.heigth = '20rem';
+    nbObjectsCanvas.width = '40rem';
+    nbObjectsCanvas.id = 'nb-objects';
+    chartDiv.appendChild(nbObjectsCanvas);
+
+    const timestamp = Date.now();
+    const nbObjectsDataset = { label: 'Number of object in Scene', data: [{ x: 0, y: 0 }] };
+    const nbObjectsChartLabel = ['0s'];
+    const nbObjectsChart = new Chart('nb-objects', {
+        type: 'line',
+        data: {
+            labels: nbObjectsChartLabel,
+            datasets: [nbObjectsDataset],
+        },
+        options: {
+            animation: { duration: 10 },
+            scales: {
+                yAxes: [{
+                    display: true,
+                    ticks: {
+                        suggestedMin: 0, // minimum will be 0, unless there is a lower value.
+                    },
+                }],
+            },
+        },
+    });
+
+    function updateNbObjectsChart() {
+        function countElem(node) {
+            if (!node) {
+                return 0;
+            }
+            let count = 1; // this node
+            if (node.children) {
+                for (const child of node.children) {
+                    count += countElem(child);
+                }
+            }
+            return count;
+        }
+
+        const timeInS = Math.floor((Date.now() - timestamp) / 1000);
+
+        const limit = 50;
+        nbObjectsChartLabel.push(`${timeInS}s`);
+        if (nbObjectsChartLabel.length > limit) {
+            nbObjectsChartLabel.shift();
+        }
+        nbObjectsDataset.data.push({ x: timeInS, y: countElem(scene.gfxEngine.scene3D) });
+        if (nbObjectsDataset.data.length > limit) {
+            nbObjectsDataset.data.shift();
+        }
+        nbObjectsChart.update();
+    }
+
+    // hook that to scene.update
+    const oldUpdate = Object.getPrototypeOf(scene).update.bind(scene);
+    scene.update = function debugUpdate() {
+        oldUpdate();
+        updateNbObjectsChart();
+    };
+
+    // DEBUG CONTROLS
+    const gui = menuGlobe.gui.addFolder('Debug Tools');
+
+    const state = {
+        showOutline: false,
+        wireframe: false,
+        displayCharts: false,
+    };
+
+    // charts
+    gui.add(state, 'displayCharts').name('Display charts').onChange((newValue) => {
+        if (newValue) {
+            chartDiv.style.display = 'flex';
+        } else {
+            chartDiv.style.display = 'none';
+        }
+    });
+
+    function applyToNodeFirstMaterial(cb) {
+        scene.getMap().tiles.children[0].traverse((object) => {
+            if (object.materials) {
+                cb(object.materials[0]);
+            }
+        });
+        scene.renderScene3D();
+    }
+
+    // tiles outline
+    gui.add(state, 'showOutline').name('Show tiles outline').onChange((newValue) => {
+        scene.map.layersConfiguration.getGeometryLayers()[0].showOutline = newValue;
```

Comment:
I've read this comment.

I'm not here to discuss the relevance of having it in the API, I'm just saying it does not belong to this PR. Why are you trying to force that on me *now*? Especially considering that

1. we clearly disagree on this matter
1. Nothing in this PR prevents these additions for later work

So there is clearly a debate, and when there is such a debate, the right thing to do is to move it on its own issue/PR. It should not prevent this one to get merged.

Anyway, because there is no consensus, and because it's not the purpose of this PR, I won't do it here, sorry.